Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/PolynomialRestShapeSpringsForceField.inl

```diff
@@ -209,7 +223,14 @@ void PolynomialRestShapeSpringsForceField<DataTypes>::addForce(const core::Mecha
 
     helper::WriteAccessor<DataVecDeriv> f1 = f;
     helper::ReadAccessor<DataVecCoord> p1 = x;
-    helper::ReadAccessor<DataVecCoord> p0 = *getExtPosition();
+
```

Comment:
is it a "probable" case that extPosition is nullptr ? In my mind, it means there is no d_restMState if m_useRestMState or mstate (if  !m_useRestMState ).
Should this test be done at the init() and set the component state at invalid instead ? (and replace the others if(!extPosition) with asserts)

Reply:
I think there is already the check in `init()`. Perhaps the component state is not correctly set (to check). But I am not fully confident in relying on the component state. Today, there is no early return based on an invalid state. To discuss if it could be generalized. And the component state value must be changed depending on the change of some Data. Which is not currently the case.
To conclude, I am not against working with component state. But:
- It is complex to handle all cases
- Need to be generalized
- I prefer to take extra measures to prevent crashes. It means checking for `nullptr` even if the component state check passed.